internal/gatewayapi/testdata/envoyproxy-endpoint-routing-disabled.in.yaml

```diff
@@ -0,0 +1,71 @@
+envoyproxy:
+  apiVersion: gateway.envoyproxy.io/v1alpha1
+  kind: EnvoyProxy
+  metadata:
+    namespace: envoy-gateway-system
+    name: test
+  spec:
+    routingType: service
+    provider:
+      type: Kubernetes
+      kubernetes:
+        envoyService:
+          type: LoadBalancer
+        envoyDeployment:
+          replicas: 2
+          container:
+            env:
+              - name: env_a
+                value: env_a_value
+              - name: env_b
+                value: env_b_name
+            image: "envoyproxy/envoy:distroless-dev"
+            resources:
+              requests:
+                cpu: 100m
+                memory: 512Mi
+            securityContext:
+              runAsUser: 2000
+              allowPrivilegeEscalation: false
+          pod:
+            annotations:
+              key1: val1
+              key2: val2
+            affinity:
+              nodeAffinity:
+                requiredDuringSchedulingIgnoredDuringExecution:
+                  nodeSelectorTerms:
+                    - matchExpressions:
+                        - key: cloud.google.com/gke-nodepool
+                          operator: In
+                          values:
+                            - router-node
+            tolerations:
+              - effect: NoSchedule
+                key: node-type
+                operator: Exists
+                value: "router"
+            securityContext:
+              runAsUser: 1000
+              runAsGroup: 3000
+              fsGroup: 2000
+              fsGroupChangePolicy: "OnRootMismatch"
+            volumes:
+              - name: certs
+                secret:
+                  secretName: envoy-cert
+gateways:
```

Comment:
suggest updating the input to also include a httproute e.g. https://github.com/envoyproxy/gateway/blob/main/internal/gatewayapi/testdata/httproute-attaching-to-gateway.in.yaml 
this will ensure we can validate the Service IP in the o/p xdsIR, currently its empty
and shaving off `envoyproxy.spec.provider` / `nil` , so this test case only focuses on `endpointType`